lib/gui/tool-runner/index.ts

```diff
@@ -321,7 +321,9 @@ export class ToolRunner {
     async run(tests: TestSpec[] = []): Promise<boolean> {
         const testCollection = this._ensureTestCollection();
 
-        return this._toolAdapter.run(testCollection, tests, this._globalOpts);
+        return _.isEmpty(tests)
```

Comment:
Hmm it's not entirely clear to me why if tests are empty here we use regular run and runWithoutRetries otherwise. We could save _.isEmpty into a boolean variable with descriptive name or leave a comment. Although I may be just missing something.

Reply:
In case when user click on `run` command in UI interface we send empty array, which means that we must run all tests with all available retires. But when user want to retry only one test or suite we disable retries.

I will add boolean variable with descriptive name.